test/parallel/test-worker-vm-context-terminate.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const vm = require('vm');
+const { Worker } = require('worker_threads');
+
+// Do not use isMainThread so that this test itself can be run inside a Worker.
+if (!process.env.HAS_STARTED_WORKER) {
+  process.env.HAS_STARTED_WORKER = 1;
+  const w = new Worker(__filename);
+  setTimeout(() => w.terminate(), 50);
```

Comment:
Is using the `'online'` event here instead an option?

Reply:
@cjihrig I’ve added that but kept the timeout, it still seems to make this fail more reliably locally (with the node `master` executable).